accepted/2024/support-nuget-authentication-plugins-dotnet-tools.md

```diff
@@ -0,0 +1,500 @@
+├   │
+│   └───CredentialProvider.Microsoft
+└───netfx
+├   ├───AWS.CodeArtifact.NuGetCredentialProvider
+├   │
+├   └───CredentialProvider.Microsoft
+└
+```
+
+The new folder structure for NuGet plugins on the Linux platforms is as follows:
+
+```log
+{user}:~/.nuget/plugins$ dir
+netcore  any
+
+{user}:~/.nuget/plugins$ cd any/
+{user}:~/.nuget/plugins/any$ dir
+botsay  dotnetsay
+
+{user}:~/.nuget/plugins/any$ ls -la
+total 164
+drwxr-xr-x 3 {user}  4096 Feb 10 08:21 .
+drwxr-xr-x 4 {user}  4096 Feb 10 08:20 ..
+drwxr-xr-x 5 {user}  4096 Feb 10 08:21 .store
+-rwxr-xr-x 1 {user} 75632 Feb 10 08:21 botsay
+-rwxr-xr-x 1 {user} 75632 Feb 10 08:20 dotnetsay
+```
+
+**Advantages:**
+
+- This approach simplifies the identification of NuGet plugins at runtime because they are installed in a location that NuGet recognizes, for example `%UserProfile%/.nuget/plugins/any` on `Windows`.
+
+**Disadvantages:**
+
+- The ideal workflow for repositories accessing private NuGet feeds, such as Azure DevOps, is to easily search for NuGet plugins and install them as global tool.
+However, this approach suggests installing the plugin as a tool-path .NET tool.
+Customers can opt to install NuGet plugins as a global tool instead of a tool-path tool.
+To do this, they need to set the `NUGET_DOTNET_TOOLS_PLUGIN_PATHS` environment variable.
+This variable should point to the location of the .NET Tool executable, which the NuGet Client tooling can invoke when needed.
+
+- This approach doesn't support developing NuGet plugins in non-.NET languages.
+
+#### Roadmap
+
+In terms of the implementation roadmap, I propose the following stages:
+
+1. Initially, plugin authors will publish NuGet plugins as .NET Tools. Consumers will install NuGet plugins using `dotnet tool` sub-commands.
+They will specify the plugin's path based on their operating system. We will make the necessary code changes on the NuGet side to enable running the plugins installed as `.NET tools`.
+
+2. Subsequently, we will introduce a `dotnet nuget plugin` subcommand. This will allow users to `install, update, uninstall, and search` for plugins.
+However, we will still rely on the `dotnet tool` command under the hood, as mentioned above.
+Please note that these command names are subject to change based on feedback to the specification we are going to create in the near future.
+
+#### Prior art
+
+- The `dotnet workload` command is separate and has its own set of sub-commands, including `install`, `uninstall`, and `list`.
+These sub-commands are wrappers for the corresponding `dotnet tool` sub-commands.
+Workloads are installed in a different location than .NET tools, which makes it easier for them to be discovered at runtime, addressing a problem that NuGet also faces.
+However, NuGet plugins and .NET tools share the similarity of being console applications.
+This prior art will be helpful for us as we consider the implementation of the `dotnet nuget plugin install/uninstall/update/search` commands.
```

Comment:
To me, the prior art section is background information that's nice to know. So, to me, there shouldn't be "prior art for the proposed design" and different "prior art for alternate design 1". It's all just one set of prior art, which is used to inform decision making.